### PR TITLE
Add validation, update recommended loading in notebook

### PR DESCRIPTION
**Pull request by cisaacstern**
@roxyboy, lots of cool developments here.

1. ORCA is online
2. I rewrote `validate_catalog.py` so that it actually validates every one of our datasets. (And running it caught a few mistakes I'd made in the catalog, which I then corrected!)
3. I then totally rewrote `intake_demo.ipynb` to take advantage of the parametrization logic in `validate_catalog.py`.

A bit more about the `intake_demo.ipynb` rewrite. Valid intake catalog parameters vary somewhat from model to model. FESOM doesn't have a `region` parameter, for example, and not all models have a `grid` on the cloud. This could be because some params are truly irrelevant for a particular model, or in certain cases it's just because we haven't gotten the data yet.

So to make it crystal clear, I've taken advantage of the existing logic in `validate_catalog.py` to import a nested dictionary into `intake_demo.ipynb` which tells you exactly the set of catalog parameters that are valid for each model. Instead of manually entering the catalog params, you can now load directly using one of these pre-defined parameterizations.

Does the new `intake_demo.ipynb` make sense? Happy to rework to make the loading more intuitive based on your feedback.

### CODE REVIEW

File: intake_demo.ipynb [hunk 1]
```diff
@@ -4,43 +4,56 @@
    "cell_type": "code",
```
Comment (roxyboy) [inline intake_demo.ipynb:4]: <p>I'm getting the error:</p><pre class="ql-syntax" spellcheck="false">ORCA36 [{'name': 'region', 'description': 'region', 'type': 'str', 'allowed': ['1', '2', '3', '4'], 'default': '1'}, {'name': 'datatype', 'description': 'surface ("surface_hourly"), interior ("interior_daily"), or grid ("grid")', 'type': 'str', 'allowed': ['surface_hourly', 'interior_daily', 'grid'], 'default': 'surface_hourly'}, {'name': 'season', 'description': 'Feb, Mar, Apr ("fma") or Aug, Sep, Oct ("aso")', 'type': 'str', 'allowed': ['fma', 'aso', ''], 'default': ''}, {'name': 'grid', 'description': 'grid type ("bathymetry" or "meshmask")', 'type': 'str', 'allowed': ['bathymetry', 'meshmask', ''], 'default': ''}]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
&lt;ipython-input-13-ad61998fbeab&gt; in &lt;module&gt;
      1 print(item, params)
----&gt; 2 ds = cat[item](**params).to_dask()
      3 ds

TypeError: ZarrSource object argument after ** must be a mapping, not list
</pre><p>when trying to run this cell...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pangeo-data/swot_adac_ogcms/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

**Comment by roxyboy** [comment]
Sorry for my belated response. I just came back from vacation.

File: intake_demo.ipynb (hunk 1 above)
Comment (cisaacstern) [reply intake_demo.ipynb:4]: @roxyboy, welcome back! Hope you had a nice vacation.

It looks like you are somehow running a mashup of the version of `intake_demo.ipynb` from the `main` branch of this repo and the version of `intake_demo.ipynb` as proposed by this PR.

The params printed out with your error message above appear to be defined like this
```python
description = cat["ORCA36"].describe()
params = description["user_parameters"]
params
```

<details>

```
[{'name': 'region',
  'description': 'region',
  'type': 'str',
  'allowed': ['1', '2', '3', '4'],
  'default': '1'},
 {'name': 'datatype',
  'description': 'surface ("surface_hourly"), interior ("interior_daily"), or grid ("grid")',
  'type': 'str',
  'allowed': ['surface_hourly', 'interior_daily', 'grid'],
  'default': 'surface_hourly'},
 {'name': 'season',
  'description': 'Feb, Mar, Apr ("fma") or Aug, Sep, Oct ("aso")',
  'type': 'str',
  'allowed': ['fma', 'aso', ''],
  'default': ''},
 {'name': 'grid',
  'description': 'grid type ("bathymetry" or "meshmask")',
  'type': 'str',
  'allowed': ['bathymetry', 'meshmask', ''],
  'default': ''}]
```

</details>

A version of ☝️ this code exists in cell 4 of https://github.com/pangeo-data/swot_adac_ogcms/blob/main/intake_demo.ipynb, but it is not present in the version of the notebook proposed by this PR, here: https://github.com/cisaacstern/swot_adac_ogcms/blob/intake-catalog/intake_demo.ipynb

**Comment by cisaacstern** [comment]
@roxyboy, I'm about to push updates to the catalog for the new FESOM data...

...so going to merge this so we can keep moving forward.

If you have lingering questions related to https://github.com/pangeo-data/swot_adac_ogcms/pull/5#discussion_r690548779 let's open a new issue to discuss.

**Pull request merged**